### PR TITLE
Closes #41 : Add ADCOEQ for review

### LINKED ISSUE #41: Template: CoEQ ADaM (ADCOEQ?)
### Background Information

Once #40 is complete we need to make a template containing the E2E derivation of ADCOEQ as detailed in the vignette.

### Definition of Done

- [ ] Create vignette
- [ ] Ensure that right at the top of the template we have text to the effect of: _UoLeeds is the license holder and any parties wishing to use the CoEQ questionnaire need to purchase the license (at the appropriate link)_ (precise wording to be agreed with UoLeeds).

### PR DESCRIPTION
**Pull request by Siddhesh2097**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Great work @Siddhesh2097 ! Just a couple of minor comments.

File: inst/templates/ad_adcoeq.R [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+# Derive four subscales
+adcoeq <- adcoeq %>%
+  call_derivation(
+    derivation = derive_summary_records,
+    variable_params = list(
+      params(
+        filter_add = PARAMCD %in% c(
+          "COEQ09", "COEQ10", "COEQ11", "COEQ12", "COEQ19"
+        ) & AVISITN >= 0,
```
Comment (starosto) [inline inst/templates/ad_adcoeq.R:136]:
```suggestion
        filter_add = PARAMCD %in% c("COEQ09", "COEQ10", "COEQ11", "COEQ12", "COEQ19" ) ,
```
why don't we calculate subscale scores for screening records?

File: inst/templates/ad_adcoeq.R [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        set_values_to = exprs(
+          AVAL = mean(AVAL, na.rm = TRUE),
```
Comment (starosto) [inline inst/templates/ad_adcoeq.R:138]: I see that we get the result with two decimal places, should we specify the number of decimal places for subscales?

File: inst/templates/ad_adcoeq.R [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+
+# Derive summary records ----
+
+# See the "Example 2(Deriving a Summary Record)" vignette section for more
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#example-2-deriving-a-summary-record)
+
+# For the Control of Eating Questionnaire, four subscales are derived.
+# These subscales are derived as the sum or the average across a subset of the
```
Comment (starosto) [inline inst/templates/ad_adcoeq.R:115]: Why do we mention "sum" here?

File: inst/templates/ad_adcoeq.R [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+# See the "Visit and Period Variables" vignette for more information:
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html)
+
+# Derive analysis visit (AVISIT, AVISITN)
+adcoeq <- adcoeq %>%
+  mutate(
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      str_detect(VISIT, "SCREEN") ~ "SCREENING",
```
[the remark this answers is not in the document]
Comment (Siddhesh2097) [reply inst/templates/ad_adcoeq.R:82]: In the vignette, BASELINE is assigned to SCREENING 1 / 2 and BASELINE , all three visits which I thought didn't make sense. So I have separated here as we collect data on screening as well.
@kathrinflunkert  @Sonali-netizen  please share your thoughts

File: inst/templates/ad_adcoeq.R (hunk 1 above)
Comment (Siddhesh2097) [reply inst/templates/ad_adcoeq.R:136]: I will again need a confirmation here from @kathrinflunkert  and @Sonali-netizen  as in what is followed

File: inst/templates/ad_adcoeq.R (hunk 3 above)
Comment (Siddhesh2097) [reply inst/templates/ad_adcoeq.R:115]: I agree, should be removed. @kathrinflunkert  Are these the only 4 parameters that we derive in COEQ ?

File: inst/templates/ad_adcoeq.R [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#example-2-deriving-a-summary-record)
+
+# For the Control of Eating Questionnaire, four subscales are derived.
+# These subscales are derived as the sum or the average across a subset of the
+# various items/questions.
+
+# The subscales are defined as follows:
+
+# 1) Craving Control: Calculate average of items 9, 10, 11, 12 and 19.
```
[the remark this answers is not in the document]
Comment (Siddhesh2097) [reply inst/templates/ad_adcoeq.R:120]: To me, both are fine, but we can be more specific.

**Comment by Siddhesh2097** [comment]
> In addition, please consider adding PARAMN, ONTRTFL, ANL01FL, CHGCAT1/N, as these are common derived variables for this type of analysis dataset.

I have added PARAMN. But other variables are project specific so I think not needed for this tempate

**Review by AndersAskeland — CHANGES_REQUESTED** [review]
Great work @Siddhesh2097 ! I only have a couple of comments.

File: inst/templates/ad_adcoeq.R [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html)
+
+# Derive analysis visit (AVISIT, AVISITN)
+adcoeq <- adcoeq %>%
+  mutate(
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      str_detect(VISIT, "SCREEN") ~ "SCREENING",
+      ADT <= TRTSDT ~ "Baseline",
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_adcoeq.R:90]: This might be a very stupid question, but would we not be able to use ```VISIT == "BASELINE" ~ "Baseline"```

And then, dependent on if we change in in the vignette, do the same there.

File: inst/templates/ad_adcoeq.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_adcoeq.R:89]: I my opinion, what makes most sense would be to have screening visits, where we don't calculate any subscales (that way we can show how to not derive summary records for specific visit), and set baseline based on VISIT. Then, we might want to change the logic a bit in the vignette to do the same.

File: inst/templates/ad_adcoeq.R (hunk 3 above)
Comment (kathrinflunkert) [reply inst/templates/ad_adcoeq.R]: Yes, the COEQ manual only mentions these 4 derived parameters.

File: inst/templates/ad_adcoeq.R (hunk 6 above)
[the remark this answers is not in the document]
Comment (Siddhesh2097) [reply inst/templates/ad_adcoeq.R:90]: ADT <= TRTSDT has been a standard when deriving baseline records. So I think we should include it and also add baseline check 
ADT <= TRTSDT & VISIT == "Baseline". This makes sense ?

File: inst/templates/ad_adcoeq.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (Siddhesh2097) [reply inst/templates/ad_adcoeq.R:89]: I agree, the current program has screening and baseline visits and we are not deriving summary scores for Screening visits using AVISITN >= 0. I will also add a comment that says we are excluding screening visits.

[the remark this answers is not in the document]
Comment (starosto) [reply inst/templates/ad_adcoeq.R:89]: @Siddhesh2097 I think the current assignment of AVISIT/N is correct here, I would rather adjust the vignette so that when we have more than one visit on or before the treatment start date, not all of them are marked as SCREENING

File: inst/templates/ad_adcoeq.R (hunk 1 above)
Comment (starosto) [reply inst/templates/ad_adcoeq.R:143]: Most likely the values from screening are not used for analysis, we can leave it as it is

File: inst/templates/ad_adcoeq.R (hunk 6 above)
[the remark this answers is not in the document]
Comment (starosto) [reply inst/templates/ad_adcoeq.R:90]: I would also rather kept it simple, so AVISIT = VISIT or AVISIT = str_to_title(VISIT)

[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_adcoeq.R:90]: I agree with @starosto . Lets keep it as simple as possible. Then people are free to implement their own logic. I will make a issue to change this in the vignette to align. So feel free to change this @Siddhesh2097 . Then we can get this merged 👍

File: inst/templates/ad_adcoeq.R (hunk 2 above)
Comment (AndersAskeland) [reply inst/templates/ad_adcoeq.R:145]: I guess this would depend on the data? Unless there is some specific requirement of how to round these subscales, I think we should allow as many decimals as `mean` wants.

File: inst/templates/ad_adcoeq.R (hunk 1 above)
Comment (AndersAskeland) [reply inst/templates/ad_adcoeq.R:143]: Agree

File: inst/templates/ad_adcoeq.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_adcoeq.R:89]: I will create a issue for changing the vignette to mirror this code.

File: inst/templates/ad_adcoeq.R [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+        set_values_to = exprs(
+          AVAL = mean(AVAL, na.rm = TRUE),
+          PARAMCD = "COEQCRSW",
+          PARAM = "COEQ - Craving for Sweet",
+          PARAMN = 23
+        )
+      ),
+      params(
+        filter_add = PARAMCD %in% c("COEQ04", "COEQ16", "COEQ17", "COEQ18") &
+          AVISIT >= 0,
```
Comment (AndersAskeland) [inline inst/templates/ad_adcoeq.R:161]:
```suggestion
          AVISITN >= 0,
```

File: inst/templates/ad_adcoeq.R (hunk 2 above)
Comment (starosto) [reply inst/templates/ad_adcoeq.R:143]: usually, when we present descriptive statistics, values calculated directly from the original value, such as mean and median, are given in the original significant decimal places + 1 additional decimal place, so I would prefer 1 decimal place in our case

**Review by AndersAskeland — CHANGES_REQUESTED** [review]
Very close to merging now. I just found this small bug :)

File: inst/templates/ad_adcoeq.R (hunk 2 above)
Comment (manciniedoardo) [reply inst/templates/ad_adcoeq.R:143]: Agree, rounding should be done in TLG programs

**Comment by Siddhesh2097** [comment]
We can merge this now 
@AndersAskeland @manciniedoardo @starosto

**Review by starosto — APPROVED** [review]

**Review by manciniedoardo — APPROVED** [review]
Great work 🥇

**Review by AndersAskeland — APPROVED** [review]
Great work!
